InnerEye/ML/run_ml.py

```diff
@@ -188,9 +191,24 @@ def setup(self, use_mount_or_download_dataset: bool = True) -> None:
             # Set local_dataset to the mounted path specified in azure_runner.py, if any, or download it if that fails
             # and config.local_dataset was not already set.
             # This must happen before container setup because that could already read datasets.
-            mounted_dataset = self.mount_or_download_dataset()
+            mounted_dataset = self.mount_or_download_dataset(self.container.azure_dataset_id,
+                                                             self.container.local_dataset)
             if mounted_dataset is not None:
                 self.container.local_dataset = mounted_dataset
+
+            if not isinstance(self.container, InnerEyeContainer):
```

Comment:
why do we need this if statement? for INnerEyeContainers, the extra... fields will simply always be empty lists?